### PR TITLE
Shapes: Optionally print shape in ocamlobjinfo

### PR DESCRIPTION
**Pull request by voodoos**
Shapes can be very verbose when printed without sharing.
This PR adds an option `-shape` to `ocamlobjinfo` so that the shape stored in a `cmt` file is printed only when explicitly requested by the user. (This PR is independent from #10825)

@Octachron this should be backported to 4.14. Do you want me to open a PR targetting that branch ? (there should be no conflicts anyway)

~~Also, does that change requires a changelog entry ?~~

### CODE REVIEW

**Review by Octachron — APPROVED** [review]
This is indeed a correct and needed refinement. Let's wait for the decision on the shape reduction merging before merging however.

**Comment by Octachron** [comment]
@voodoos , I will merge as soon as you update the Change entry.

**Comment by voodoos** [comment]
@Octachron done !

**Pull request merged**
